core/backend/src/workspace/Settings.ts

```diff
@@ -86,237 +97,91 @@ export interface Settings {
    * if its type is correct.
    */
   getSetting<T extends SettingType>(settingName: SettingName, defaultValue?: T): T | undefined;
 
+  getSettingEntries<T extends SettingType>(settingName: SettingName): Iterable<{ value: T, dictionary: Settings.Dictionary}>;
+
+  getSettingValues<T extends SettingType>(settingName: SettingName): Iterable<T>;
+  
   /** Get a string setting by SettingName.
    * @param settingName The name of the setting
-   * @param defaultValue value returned if settingName is not present in any SettingDictionary, or if the highest priority setting is not a string.
+   * @param defaultValue value returned if settingName is not present in any Settings.Dictionary, or if the highest priority setting is not a string.
    */
   getString(settingName: SettingName, defaultValue: string): string;
   getString(settingName: SettingName, defaultValue?: string): string | undefined;
 
   /** Get a boolean setting by SettingName.
   * @param settingName The name of the setting
-  * @param defaultValue value returned if settingName is not present in any SettingDictionary, or if the highest priority setting is not a boolean.
+  * @param defaultValue value returned if settingName is not present in any Settings.Dictionary, or if the highest priority setting is not a boolean.
   */
   getBoolean(settingName: SettingName, defaultValue: boolean): boolean;
   getBoolean(settingName: SettingName, defaultValue?: boolean): boolean | undefined;
 
   /** Get a number setting by SettingName.
   * @param settingName The name of the setting
-  * @param defaultValue value returned if settingName is not present in any SettingDictionary, or if the highest priority setting is not a number.
+  * @param defaultValue value returned if settingName is not present in any Settings.Dictionary, or if the highest priority setting is not a number.
   */
   getNumber(settingName: SettingName, defaultValue: number): number;
   getNumber(settingName: SettingName): number | undefined;
 
   /** Get an object setting by SettingName.
   * @param settingName The name of the setting
-  * @param defaultValue value returned if settingName is not present in any SettingDictionary, or if the highest priority setting is not an object.
+  * @param defaultValue value returned if settingName is not present in any Settings.Dictionary, or if the highest priority setting is not an object.
   */
   getObject<T extends object>(settingName: SettingName, defaultValue: T): T;
   getObject<T extends object>(settingName: SettingName): T | undefined;
 
   /** Get an array setting by SettingName.
   * @param settingName The name of the setting
-  * @param defaultValue value returned if settingName is not present in any SettingDictionary, or if the highest priority setting is not an array.
+  * @param defaultValue value returned if settingName is not present in any Settings.Dictionary, or if the highest priority setting is not an array.
```

Comment:
I just noticed that these comments are now wrong. It throws if the value doesn't match.

Reply:
Updated